### PR TITLE
Flake8 Changes + Fixing Warnings

### PR DESCRIPTION
**Pull request by GameDungeon**
## Overview: What does this pull request change?

Removed almost all of the warnings from pytest. Readded the flake8 plugins removed in #1323 for unknown reasons. Minor pre-commit changes to help with flake8 compliance.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Comment by GameDungeon** [comment]
I do know this is nearly impossible to properly review. It's just some minor changes to code to remove flake8 ignores. The bulk of this is the fact that the trailing whitespace pre-commit hook got out of hand.

File: manim/renderer/webgl_renderer.py
Comment (hydrobeam) [inline manim/renderer/webgl_renderer.py:63]: What's the reasoning behind using a dataclass here?

File: manim/utils/config_ops.py
Comment (hydrobeam) [inline manim/utils/config_ops.py:45]: same question as before

File: tests/test_hashing.py
Comment (hydrobeam) [inline tests/test_hashing.py:12]: What does this change do 🤔

File: tests/test_texmobject.py
Comment (hydrobeam) [inline tests/test_texmobject.py:18]: what does this comment mean?

**Review by hydrobeam — APPROVED** [review]
I looked through all the changes and they seem good to me. Sometimes the trailing commas that were added felt a bit odd, but it's not too concerning. I liked the cases where arguments were split into multiple lines since it makes the code easier to parse 👍. Plus, the actual code changes in this PR improve readability + code quality, so that's nice too.

I left some questions about some of the changes below.

**Comment by behackl** [comment]
Should we be able to resolve all of the remaining open discussions here in time, I propose to merge this right after the release. (This almost certainly introduces merge conflicts with other PRs, which is inconvenient for the last few hours of work before the release.)

File: manim/mobject/mobject.py
Comment (GameDungeon) [reply manim/mobject/mobject.py:600]: It was adding 20k warnings. It gets to the point where it would cover up any warnings that we may want to know about.

File: manim/renderer/webgl_renderer.py
Comment (GameDungeon) [reply manim/renderer/webgl_renderer.py:63]: Flake was telling me to, so I started fixing that ignore. I found it was not worth it and just ignored it. I can remove the dataclass if you want, though this is the exact case dataclass is designed for.

File: tests/test_hashing.py
Comment (GameDungeon) [reply tests/test_hashing.py:12]: Here I just removed an unneeded arg as scope="function" is the deafult.

File: tests/test_texmobject.py
Comment (GameDungeon) [reply tests/test_texmobject.py:18]: That's supposed to be a noqa. Flake had a weird error here.

**Comment by GameDungeon** [comment]
Ok, remind me after the release to update the pull to the latest version.

File: manim/mobject/mobject.py
Comment (hydrobeam) [reply manim/mobject/mobject.py:600]: What are the warnings this fixes? I believe there might be another way around them.

Comment (GameDungeon) [reply manim/mobject/mobject.py:600]: > manim\manim\mobject\value_tracker.py:172: DeprecationWarning: This method is not guaranteed to stay around. Please prefer getting the attribute normally.

26048 warnings. I asked in manim dev, and the person that added said warnings told me to do this.

File: manim/renderer/webgl_renderer.py
Comment (hydrobeam) [reply manim/renderer/webgl_renderer.py:63]: It doesn't feel related to this PR (and isn't implemented properly as it's overwriting the `__init__` a dataclass would generate) so I think it'd be better to remove it.

Comment (GameDungeon) [reply manim/renderer/webgl_renderer.py:63]: ok, I'll get on that

**Review by hydrobeam — CHANGES_REQUESTED** [review]
Switching to `Request Changes` since I'm not a fan of adding code just to appease warnings. PR's such as #1926 should make it possible to do away with the warning code properly.

**Comment by GameDungeon** [comment]
I did the requested changes. 
I would prefer to do this merge sooner rather than later due to the merge conflicts it would likely cause.

**Comment by GameDungeon** [comment]
Thanks :)

File: .flake8
Comment (hydrobeam) [inline .flake8:8]: Could you add some explanation via a comment as to what the purpose of these lines are?

File: .flake8
Comment (hydrobeam) [inline .flake8:2]: Why remove this comment 🤔

**Review by hydrobeam — COMMENTED** [review]
Reviewing the changes again before merging, I had some questions about some of the chagnes in the .flake8 file.

File: .flake8
Comment (GameDungeon) [reply .flake8:2]: That was a mistake, I can add it back.

File: .flake8
Comment (GameDungeon) [reply .flake8:8]: Sure

**Comment by hydrobeam** [comment]
Thanks!

**Pull request merged**
